zebra-chain/src/block/difficulty.rs

```diff
@@ -93,6 +93,45 @@ impl fmt::Debug for ExpandedDifficulty {
     }
 }
 
+/// A 128-bit unsigned "Work" value.
+///
+/// Used to calculate the total work for each chain of blocks.
+///
+/// Details:
+///
+/// The relative value of `Work` is consensus-critical, because it is used to
+/// choose the best chain. But its precise value and bit pattern are not
+/// consensus-critical.
+///
+/// We calculate work values according to the Zcash specification, but store
+/// them as u128, rather than the implied u256. We don't expect the total chain
+/// work to ever exceed 2^128. The current total chain work for Zcash is 2^58,
+/// and Bitcoin adds around 2^91 work per year. (Each extra bit represents twice
+/// as much work.)
```

Comment:
If my rough sums are correct, it would take over a million times earth's current energy usage to do this much work. Or about a thousand times the energy earth gets from the sun.

So yeah. 2^64 is a bit risky, 2^128 is way too big.